EarlGrey/Event/GREYSyntheticEvents.m

```diff
@@ -79,106 +79,154 @@ + (void)touchAlongPath:(NSArray *)touchPath
+                                      expendable:expendable];
+    }
 
-  for (NSUInteger i = 1; i < [touchPath count]; i++) {
-    [eventGenerator grey_continueTouchAtPoint:[touchPath[i] CGPointValue]
-        afterTimeElapsedSinceLastTouchDelivery:delayBetweenEachEvent
-                             immediateDelivery:NO
-                                    expendable:expendable];
+    [eventGenerator grey_endTouchesAtPoints:[self grey_objectsAtIndex:firstTouchPathSize - 1
+                                                             ofArrays:touchPaths]
+          timeElapsedSinceLastTouchDelivery:0];
   }
-  [eventGenerator endTouch];
 }
 
 - (void)beginTouchAtPoint:(CGPoint)point
          relativeToWindow:(UIWindow *)window
         immediateDelivery:(BOOL)immediate {
-  NSAssert(!_touchInjector, @"Cannot call this method more than once until endTouch is called.");
-
-  _touchInjector = [[GREYSingleSequenceTouchInjector alloc] initWithWindow:window];
-  GREYTouchInfo *touchInfo = [[GREYTouchInfo alloc] initWithPoint:point
-                                                        lastTouch:NO
-                                  deliveryTimeDeltaSinceLastTouch:0
-                                                       expendable:NO];
-  [_touchInjector enqueueTouchInfoForDelivery:touchInfo];
-  if (immediate) {
-    [self grey_waitForInjectorToFinishDeliveringTouches:_touchInjector];
-  }
+  [self grey_beginTouchesAtPoints:@[[NSValue valueWithCGPoint:point]]
+                 relativeToWindow:window
+                immediateDelivery:immediate];
 }
 
 - (void)continueTouchAtPoint:(CGPoint)point
            immediateDelivery:(BOOL)immediate
                   expendable:(BOOL)expendable {
-  [self grey_continueTouchAtPoint:point
+  [self grey_continueTouchAtPoints:@[[NSValue valueWithCGPoint:point]]
       afterTimeElapsedSinceLastTouchDelivery:0
                            immediateDelivery:immediate
                                   expendable:expendable];
 }
 
 - (void)endTouch {
-  GREYTouchInfo *touchInfo = [[GREYTouchInfo alloc] initWithPoint:CGPointZero
-                                                        lastTouch:YES
-                                  deliveryTimeDeltaSinceLastTouch:0
-                                                       expendable:NO];
-  [_touchInjector enqueueTouchInfoForDelivery:touchInfo];
-  [self grey_waitForInjectorToFinishDeliveringTouches:_touchInjector];
-
-  _touchInjector = nil;
+  [self grey_endTouchesAtPoints:@[[NSValue valueWithCGPoint:CGPointZero]]
+      timeElapsedSinceLastTouchDelivery:0];
 }
 
 #pragma mark - Private
 
+// Given an array containing multiple arrays, returns an array with the index'th element of each
+// array.
++ (NSArray *)grey_objectsAtIndex:(NSUInteger)index ofArrays:(NSArray *)arrayOfArrays {
+  NSAssert([arrayOfArrays count] > 0, @"arrayOfArrays must contain at least one element.");
+
+  NSUInteger firstArraySize = [arrayOfArrays[0] count];
```

Comment:
Change this to `GREY_UNUSED_VARIABLE NSUInteger firstArraySize = [arrayOfArrays[0] count];` to prevent Carthage warning.